### ISSUE TITLE
Change Shade Groups of Apertures and Faces to be Assigned to the Space and not the Building

### ISSUE THREAD

**Issue by chriswmackey** [ladybug-tools-in2/energy-model-measure#29]
This may be a bug in the OpenStudio SDK but I am going to post here first until I can verify for sure that it's not an issue with the energy-model-measure.  Specifically, it seems that, when a Construction is assigned to SpaceShading in a ConstructionSet:
![image](https://user-images.githubusercontent.com/5567574/71527747-b980d180-28aa-11ea-9863-a83f493184b5.png)
![image](https://user-images.githubusercontent.com/5567574/71527770-d0bfbf00-28aa-11ea-82a7-07bf88e37a4b.png)

It actually writes a real construction into the IDF, which cannot be assigned to shades according to the EnergyPlus Input/Output reference:
![image](https://user-images.githubusercontent.com/5567574/71527728-a968f200-28aa-11ea-945b-625808b6fe59.png)

Also, I can verify that there are no ShadingProperty:Reflectance objects in the resulting IDF, which I think it want OpenStudio is supposed to do according to the code here:
https://github.com/NREL/OpenStudio/blob/develop/src/energyplus/ForwardTranslator/ForwardTranslateShadingSurface.cpp#L173-L235

Because things seem correct in the OSM but not in the IDF, I am thinking this is probably a bug in the OpenStudio SDK but I would like to get @DavidGoldwasser and @tanushree04 's thoughts first.

**Comment by DavidGoldwasser** [comment]
@chriswmackey I believe you are correct that EnergyPlus doesn't have a construction for shading surfaces. I believe that the construction was used for Radiance translation, but did have some functionality related to reflectance in EnergyPlus. 

Looking at Input Output ref here, have you confirmed the solar reflectance field in the building object is FullExteriorWithReflections or FullInteriorAndExteriorWithReflections? It is possible that impacts if the object is written.
https://bigladdersoftware.com/epx/docs/9-2/input-output-reference/group-thermal-zone-description-geometry.html#shading-surfaces

I'll take a look at that code and I'll do some tests on a clean model with hard assigned and default construction assignment.

**Comment by chriswmackey** [comment]
@DavidGoldwasser ,

Yes, EnergyPlus definitely does not have a construction for Shading surfaces as of the time I am writing this.

Right now, the only way to change the properties of a Shading surface in EnergyPlus is through the [ShadingProperty:Reflectance](https://bigladdersoftware.com/epx/docs/9-2/input-output-reference/group-thermal-zone-description-geometry.html#shadingpropertyreflectance) (or through a transmittance schedule but I'm not worried about that right now).  You can see that the .cpp code that I linked to there seems to be translating the OpenStudio-assigned construction to an EnergyPlus ShadingProperty:Reflectance but, for some reason, this translation is not happening in the test I have run on my end.  I can also confirm that I am using `FullInteriorAndExteriorWithReflections`.

Here is an OpenStudio Model that you can run through the ForwardTranslator and see that I am not getting a ShadingProperty:Reflectance, even though there is a shade material assigned to the ConstructionSet:
https://drive.google.com/file/d/11C6nsH7TNe2Dg3eFErijnmZZ-Zfq15Fd/view?usp=sharing

![image](https://user-images.githubusercontent.com/5567574/71593292-77050200-2b01-11ea-9a40-6af71099e746.png)

**Comment by DavidGoldwasser** [comment]
@chriswmackey I think it is just an input error on the model. While the default construction set you show above is assigned to the facility as the default construction set. The space in the model has overridden that default to use the 2013 construction set, which does not include a space shading construction. 

Although I did think you could nest construction sets, as an example that a default construction set is complete, but a first story construction set could just have the exterior wall set (for example to stone vs. stucco). That would seem to apply in this case but appears not to. Unless something else is going on.

I did make clean test model with space, building, and site shading with a default construction, and all of them did work. Anyway so the easy fix now, but be to not rely on two interlinked construction sets. I'll look into that more. below for reference is my test.

![Screen Shot 2019-12-30 at 3 45 07 PM](https://user-images.githubusercontent.com/4857055/71603948-8cd3f080-2b1c-11ea-9de5-b71ad962947a.png)

**Comment by DavidGoldwasser** [comment]
Hmm, no, on my test model the inheritance works, something else is going on in your model. I'll reply later when I figure it out. I can see the same issue you do with your model in the ShadingProperty not being made

**Comment by DavidGoldwasser** [comment]
Ah, all the shading surface groups are assigned to the building, not the space. Also, I did see a number of them don't have surfaces assigned to them. That won't cause a failure anywhere, but isn't ideal.

<img width="1680" alt="Screen Shot 2019-12-30 at 4 33 52 PM" src="https://user-images.githubusercontent.com/4857055/71604820-d07d2900-2b21-11ea-874f-174d7c36387b.png">

**Comment by chriswmackey** [comment]
Thank you for getting to the bottom of this @DavidGoldwasser and sorry to have troubled you about something that isn't an issue with the OpenStudio SDK. In this case, it looks like it's just an issue with [this part of the measure](https://github.com/ladybug-tools-in2/energy-model-measure/blob/master/lib/ladybug/energy_model/aperture.rb#L82-L89). We probably should also fix [this part of the measure](https://github.com/ladybug-tools-in2/energy-model-measure/blob/master/lib/ladybug/energy_model/face.rb#L135-L142). However, [this part of the measure](https://github.com/ladybug-tools-in2/energy-model-measure/blob/master/lib/ladybug/energy_model/room.rb#L125-L133). looks like it is correct and so we don't need to change it.

Basically, if we assign each Shade group to the Space that it is a part of in these places, then all should work.

@tanushree04 , do you think this might be a bug that you could fix before you set off for your vacation? If not, I can try my hand at fixing it.

**Comment by chriswmackey** [comment]
Also @tanushree04 , the reason why we have shading groups that possess no shades is that you are automatically making a surface group for every Face and Aperture in the model [here](https://github.com/ladybug-tools-in2/energy-model-measure/blob/master/lib/ladybug/energy_model/face.rb#L133) and [here](https://github.com/ladybug-tools-in2/energy-model-measure/blob/master/lib/ladybug/energy_model/aperture.rb#L80). You need to put this code inside the `if` statement and also do a check to be sure that the length of the `otudoor_shades` isn't 0 before you go and create the Shade Groups.

**Comment by DavidGoldwasser** [comment]
No problem @chriswmackey happy new year a few hours early!

**Comment by tanushree04** [comment]
@DavidGoldwasser @chriswmackey Thank you for pointing out what needs to be done, I can take care of it before I leave.

**Comment by chriswmackey** [comment]
I pushed a change that generally fixes this:
https://github.com/ladybug-tools-in2/energy-model-measure/pull/35
